node_shims.ts

```diff
@@ -0,0 +1,25 @@
+if (!("ReadableStream" in globalThis) || !("TransformStream" in globalThis) || !("WritableStream" in globalThis)) {
+  (async () => {
+    const { ReadableStream, TransformStream, WritableStream } = await import("node:stream/web");
+    Object.defineProperties(globalThis, {
+      "ReadableStream": {
+        value: ReadableStream,
+        writable: true,
+        enumerable: false,
+        configurable: true,
+      },
+      "TransformStream": {
+        value: TransformStream,
+        writable: true,
+        enumerable: false,
+        configurable: true,
+      },
+      "WritableStream": {
+        value: WritableStream,
+        writable: true,
+        enumerable: false,
+        configurable: true,
+      },
+    });
+  })();
+}
```

Comment:
I think this ^ merging into globals will be done automatically by `dnt` if you specify it in the DNT config `shims.custom`:

```
  shims: {
    deno: {
      test: "dev",
    },
    custom: [{
      package: {
        name: "node:stream/web",
      },
      globalNames: [
        "ReadableStream",
        "WritableStream",
        "TransformStream",
      ],
    }],
  },
```

See https://github.com/denoland/dnt/issues/362 and [this integration test which checks that this approach works](https://github.com/denoland/dnt/blob/2a144787a696cdb6bf3488358edf5f76bd296767/rs-lib/tests/integration_test.rs#L327-L361).

Reply:
I did it [here](https://github.com/bradenmacdonald/deno-s3-lite-client/pull/30/commits/811be45b04d1262641866bbb27903ed20a8bd5b1), and it does work on Node.js env but break in Workers env.
That why I implemented as oak project does (made by a Deno maintainer)